lms/models/canvas_file.py

```diff
@@ -0,0 +1,44 @@
+import sqlalchemy as sa
+from sqlalchemy.dialects.postgresql import ARRAY
+from sqlalchemy.ext.mutable import MutableList
+
+from lms.db import BASE
+
+
+class CanvasFile(BASE):
+    """
+    A file that we've seen in a response from the Canvas Files API.
+
+    https://canvas.instructure.com/doc/api/files.html
+    """
+
+    __tablename__ = "canvas_file"
+    __table_args__ = (
+        sa.UniqueConstraint("consumer_key", "tool_consumer_instance_guid", "file_id"),
+    )
+
+    id = sa.Column(sa.Integer, primary_key=True)
+
+    consumer_key = sa.Column(
+        sa.String,
+        sa.ForeignKey("application_instances.consumer_key", ondelete="cascade"),
+        nullable=False,
+    )
+    tool_consumer_instance_guid = sa.Column(sa.UnicodeText, nullable=False)
+    file_id = sa.Column(sa.Integer, nullable=False)
```

Comment:
Canvas file IDs are global within the Canvas instance (they're not per-course IDs).

Each `file_id` in our DB model is namespaced by:

* `tool_consumer_instance_guid`: The LTI 1 launch param that uniquely identifies the Canvas instance. Because different Canvas instances will use the same `file_id` for different files
* `consumer_key`: The LTI 1 launch param (`oauth_consumer_key`) that uniquely identifies the application instance that was active when this file record was launched. This is necessary because the consumer key and secret are the basis of security in LTI 1. If we didn't namespace by consumer key then I could register an application instance using your `tool_consumer_instance_guid` and use a fake LTI launch and fake Canvas API to poison all your Canvas file records in the DB (e.g. rename all the files)
* Note that it's not enough to namespace by `consumer_key` only since a single application instance can be installed in multiple Canvas instances